src/libs/Localize/index.ts

```diff
@@ -58,16 +81,43 @@ type Phrase<TKey extends TranslationPaths> = TranslationFlatObject[TKey] extends
 function translate<TKey extends TranslationPaths>(desiredLanguage: 'en' | 'es' | 'es-ES' | 'es_ES', phraseKey: TKey, ...phraseParameters: PhraseParameters<Phrase<TKey>>): string {
     // Search phrase in full locale e.g. es-ES
     const language = desiredLanguage === CONST.LOCALES.ES_ES_ONFIDO ? CONST.LOCALES.ES_ES : desiredLanguage;
+
+    // Get the cache for the above locale
+    let cacheForLocale = translationCache.get(language);
+
+    // Directly access and assign the translated value from the cache, instead of
+    // going through map.has() and map.get() to avoid multiple lookups.
+    const valueFromCache = cacheForLocale?.get(phraseKey);
+
+    // If the phrase is already translated, return the translated value
+    if (valueFromCache) {
+        return valueFromCache;
+    }
+
     let translatedPhrase = translations?.[language]?.[phraseKey] as Phrase<TKey>;
```

Comment:
> If that's the case, why don't we modify the structure of the translations to return a map instead of a plain object in this file?

That makes sense to me. I'm also a bit confused by why `translate` is slow, since it should just be a direct lookup in an object?

Reply:
@fedirjh I will look into this today and see whether changing translation structure to Map performs better than what we have now 👍

Reply:
@fedirjh @roryabraham  Here is an update on how things looks like with:

- translations with object structure ( what we have in main )
- translations with Map structure ( the proposed change by @fedirjh  )
- translations with object structure with cache applied ( what we have right now in this PR )

Below is a video attached which shows the Hermes traces on a heavy account ( Release + Android real device ). For the same section of the trace, we observe that:

- translations with object structure ( before translations Tab in video ) has 11 calls to translate method
- translations with map structure ( after translations with Map Tab in video ) has 5 calls to translate method
- translations with object structure with cache applied ( after translations with cache Tab in video ) has 3 calls to translate method

https://github.com/Expensify/App/assets/47336142/9cfa68cb-7d61-4ce4-be02-f61883331f09


One thing to note here is that "I only created a Map of `en` translations with only 300 entries, whereas we have around 2000 entries in object structure." to test the map structure but since it had more calls than cache implemented, it's safe to say that it's not worth to change object to map.

Below is another video, where we can see the cache implemented and see that how continuously we are accessing the huge translations object to get the desired value for the key. I just want to emphasise that since the cache Map is marginally small as compared to the translations object and that's why we have so less recordings of the `translate` function in Hermes trace, when cache is enabled. It's to note that the `translate` function will still be called but since we are returning from cache and it's quick enough to not get caught in the Hermes trace and the times it get caught was 3.

https://github.com/Expensify/App/assets/47336142/29ea54b5-66ed-4ed8-87c4-9d7607c99860